### ISSUE TITLE
Add `PropertyNamingStrategyWrapper` support for `PropertyNamingStrategy`s that extend `NamingBase`

### ISSUE THREAD

**Issue by bobtiernay-okta** [HubSpot/jackson-datatype-protobuf#81]
Currently this uses reflection that doesn't work well with later versions of Jackson (e.g. `2.12.3`) which added [new implementations](https://github.com/FasterXML/jackson-databind/issues/2715):

```java
  public PropertyNamingStrategyWrapper(PropertyNamingStrategy delegate) {
    // Doesn't support `NamingBase` decendants!
    if (delegate instanceof PropertyNamingStrategyBase) {
      this.delegate = (PropertyNamingStrategyBase) delegate;
    } else if (delegate == PropertyNamingStrategy.LOWER_CAMEL_CASE) {
      this.delegate = NO_OP;
    } else {
      this.delegate = SNAKE_TO_CAMEL;
    }
  }
```

So if this is used:

```java
var mapper = new ObjectMapper().registerModule(new ProtobufModule()).setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE);
var value = mapper.convertValue(...);
```

will incorrectly result in `value` to have `SNAKE_TO_CAMEL` being applied.

**Comment by jhaber** [comment]
Should be fixed by #88

**Comment by michielboekhoff** [comment]
Could this please be released?
